xds/googledirectpath/googlec2p.go

```diff
@@ -135,6 +137,15 @@ func (c2pResolverBuilder) Build(t resolver.Target, cc resolver.ClientConn, opts
 
 	// Create and return an xDS resolver.
 	t.Scheme = xdsName
```

Comment:
This line seems redundant now given that we are assigning a totally new value to `t` in line 141.

Also, `t.Scheme` is marked deprecated in the docs. Should we do something similar to what is being done here in line 95 as well (so that the correct scheme is set in the `Scheme` field inside the `URL`).

Reply:
> This line seems redundant now given that we are assigning a totally new value to t in line 141.

Note this field is assigned a new value only if federation is enabled.

> Also, t.Scheme is marked deprecated in the docs. Should we do something similar to what is being done here in line 95 as well (so that the correct scheme is set in the Scheme field inside the URL).

I was actually a bit confused about what this code needs to handle, with the non-URL fields being marked deprecated.

- Should we assume that the `resolver.Target` param has a valid URL field ?
- Do we need to set both `resolver.Target.URL.Scheme and  `resolver.Target.Scheme` ?

Reply:
FWIW, the current approach just seemed the simplest thing to do to get tests to pass.

Reply:
> Should we assume that the resolver.Target param has a valid URL field ?

Yes, as part of channel creation, we parse the target URI supplied by the user and stored the parsed URL in the `URL` field of `resolver.Target`. https://github.com/grpc/grpc-go/blob/master/clientconn.go#L1587-L1611.

> Do we need to set both `resolver.Target.URL.Scheme` and `resolver.Target.Scheme`

The `resolver.Target.URL` field was added when we cleaned up the mess around target URI parsing. But we could not get rid of the existing fields because of backward compatibility reasons. But, ideally the values for the old fields and `resolver.Target.URL` should be compatible.

So, if we are changing one of them, it would be nice to ensure that the other one is in sync.

Reply:
Updated to set both the old and new versions of the `Scheme` field. PTAL